### PR TITLE
fix: rename ErrSkipDesc to SkipNode

### LINKED ISSUE #634: Rename `ErrSkipDesc` to `SkipNode`
#631 exports `errSkipDesc` to `ErrSkipDesc` so that copy can be skipped for mounting. However, `ErrSkipDesc` is not a real error that should be returned by any function. 

I have looked into the golang built-in package [`io/fs`](https://pkg.go.dev/io/fs#pkg-variables) and found `SkipAll` and `SkipDir`. To have consistent naming as the build-in packages, I propose that we should rename `ErrSkipDesc` to ~~`SkipDesc`~~ `SkipNode` (as discussed in comment section).

```go
var SkipAll = errors.New("skip everything and stop the walk")
var SkipDir = errors.New("skip this directory")
```

@Wwwsylvia @ktarplee Any comments?

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Resolves #634

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: copy.go [hunk 1]
```diff
@@ -37,9 +37,9 @@ import (
 // defaultConcurrency is the default value of CopyGraphOptions.Concurrency.
 const defaultConcurrency int = 3 // This value is consistent with dockerd and containerd.
 
-// ErrSkipDesc signals to stop copying a descriptor.  When returned from PreCopy the blob must exist in the target.
+// SkipNode signals to stop copying a descriptor.  When returned from PreCopy the blob must exist in the target.
 // This can be used to signal that a blob has been made available in the target repository by "Mount()" or some other technique.
-var ErrSkipDesc = errors.New("skip descriptor")
+var SkipNode = errors.New("skip descriptor")
```
Comment (ktarplee) [inline copy.go:42]: I think we should change the text of the error to "skip node".

**Comment by ktarplee** [comment]
I also think we should add a sentence to the documentation for the `PreCopy` option that describes what happens when that function returns a SkipNode error.

File: copy.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply copy.go:42]: changed the error text to "skip node".

**Comment by wangxiaoxuan273** [comment]
> I also think we should add a sentence to the documentation for the `PreCopy` option that describes what happens when that function returns a SkipNode error.

Added a doc comment in the `PreCopy` field.

File: copy.go [hunk 2]
```diff
@@ -96,7 +96,10 @@ type CopyGraphOptions struct {
 	// cached in the memory.
 	// If less than or equal to 0, a default (currently 4 MiB) is used.
 	MaxMetadataBytes int64
-	// PreCopy handles the current descriptor before copying it.
+	// PreCopy handles the current descriptor before copying it. It returns an
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply copy.go:99]: changed accordingly.

File: copy.go [hunk 3]
```diff
@@ -37,9 +37,9 @@ import (
 // defaultConcurrency is the default value of CopyGraphOptions.Concurrency.
 const defaultConcurrency int = 3 // This value is consistent with dockerd and containerd.
 
-// ErrSkipDesc signals to stop copying a descriptor.  When returned from PreCopy the blob must exist in the target.
+// SkipNode signals to stop copying a descriptor. When returned from PreCopy the blob must exist in the target.
```
[the remark this answers is not in the document]
Comment (wangxiaoxuan273) [reply copy.go:40]: changed accordingly.

**Review by Wwwsylvia — APPROVED** [review]
LGTM

**Pull request merged**